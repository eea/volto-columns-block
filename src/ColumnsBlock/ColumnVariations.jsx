import React from 'react';
import { Segment, Card } from 'semantic-ui-react';
import { Icon } from '@plone/volto/components';

export default ({ data, onChange, children, variants }) => {
  // data by default is like { '@type': 'columns }
  return Object.keys(data).length === 1 ? (
    <Segment>
      <Card.Group>
        {variants.map(({ icon, defaultData, title }, index) => (
          <Card key={index} onClick={() => onChange(defaultData)}>
            <Card.Content>
              <Icon name={icon} />
              {title ? <Card.Header>{title}</Card.Header> : ''}
            </Card.Content>
          </Card>
        ))}
      </Card.Group>
    </Segment>
  ) : (
    children
  );
};
