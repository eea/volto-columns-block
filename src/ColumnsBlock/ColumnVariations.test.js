import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';

import ColumnVariations from './ColumnVariations';

const mockStore = configureStore();

test('renders column variations component', () => {
  const store = mockStore({
    intl: {
      locale: 'en',
      messages: {},
    },
  });

  const component = renderer.create(
    <Provider store={store}>
      <ColumnVariations
        id="column-variations"
        title="Column variations"
        fieldSet="default"
        variants={[
          {
            icon: {},
            defaultData: {
              gridSize: 12,
              gridCols: ['halfWidth', 'halfWidth'],
            },
            common: true,
            title: '50 / 50',
          },
        ]}
        onChange={() => {}}
        onClick={() => {}}
      />
    </Provider>,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
