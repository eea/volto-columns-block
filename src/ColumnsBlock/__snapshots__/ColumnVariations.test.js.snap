// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders column variations component 1`] = `
<div
  className="ui segment"
>
  <h4>
    Select layout:
  </h4>
  <div
    className="ui centered seven cards"
  >
    <a
      className="ui card"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <svg
          className="icon"
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
          onClick={null}
          style={
            Object {
              "fill": "currentColor",
              "height": "55px",
              "width": "auto",
            }
          }
        />
        <p>
          50 / 50
        </p>
      </div>
    </a>
  </div>
</div>
`;
